File: src/main/java/recipeapplication/security/UserDto.java
package recipeapplication.security;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

public class UserDto
{
    @NotNull
    @NotEmpty
    private String username;

    @NotNull
    @NotEmpty
    private String password;

    @NotNull
    @NotEmpty
    private String email;

    public String getUsername()
    {
        return username;
    }

    public void setUsername(String username)
    {
        this.username = username;
    }

    public String getPassword()
    {
        return password;
    }

    public void setPassword(String password)
    {
        this.password = password;
    }

    public String getEmail()
    {
        return email;
    }

    public void setEmail(String email)
    {
        this.email = email;
    }
}

File: src/main/kotlin/thecookingpot/dto/UserDto.kt
package thecookingpot.dto

import javax.validation.constraints.Email
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.Size

class UserDto {
    lateinit var username: @NotEmpty @Size(min = 3) String
    lateinit var password: @NotEmpty @Size(min = 5) String
    lateinit var email: @NotEmpty @Email String
}